pkg/zk/generators.go

```diff
@@ -239,14 +261,15 @@ func makeService(name string, ports []v1.ServicePort, clusterIP bool, z *v1beta1
 					Kind:    "ZookeeperCluster",
 				}),
 			},
-			Labels: map[string]string{"app": z.GetName()},
+			Labels:      map[string]string{"app": z.GetName()},
+			Annotations: annotationMap,
 		},
 		Spec: v1.ServiceSpec{
 			Ports:    ports,
 			Selector: map[string]string{"app": z.GetName()},
 		},
 	}
-	if clusterIP == false {
+	if dnsEnabled || clusterIP == false {
```

Comment:
Could we just say `if ( dnsEnabled || !clusterIP )` its easier to read.

Reply:
I don't see the point in setting **ClusterIP=None** for the `zk client service` as that essentially makes it a headless service. We already have a headless service for zk cluster.
Is the annotation expected be added to the headless service??

Reply:
As per the discussion with Renjith, added the annotation to the zk headless service.